bolt-cli/src/commands/operators.rs

```diff
@@ -149,23 +149,37 @@ impl OperatorsCommand {
                         Bytes::from(signer.sign_hash_sync(&signature_digest_hash)?.as_bytes());
                     let signature = SignatureWithSaltAndExpiry { signature, expiry, salt };
 
-                    let result = bolt_eigenlayer_middleware
+                    match bolt_eigenlayer_middleware
                         .registerOperator(operator_rpc.to_string(), signature)
                         .send()
-                        .await?;
-
-                    info!(
-                        hash = ?result.tx_hash(),
-                        "registerOperator transaction sent, awaiting receipt..."
-                    );
-
-                    let receipt = result.get_receipt().await?;
-                    if !receipt.status() {
-                        eyre::bail!("Transaction failed: {:?}", receipt)
+                        .await
+                    {
+                        Ok(pending) => {
+                            info!(
+                                hash = ?pending.tx_hash(),
+                                "registerOperator transaction sent, awaiting receipt..."
+                            );
+
+                            let receipt = pending.get_receipt().await?;
+                            if !receipt.status() {
+                                eyre::bail!("Transaction failed: {:?}", receipt)
+                            }
+
+                            info!("Succesfully registered EigenLayer operator");
+                        }
+                        Err(e) => {
+                            match try_parse_contract_error::<BoltEigenLayerMiddlewareErrors>(e)? {
+                                BoltEigenLayerMiddlewareErrors::AlreadyRegistered(_) => {
+                                    eyre::bail!("Operator already registered in bolt")
+                                }
+                                BoltEigenLayerMiddlewareErrors::NotOperator(_) => {
+                                    eyre::bail!("Operator not registered in EigenLayer")
+                                }
+                                _ => unreachable!(),
```

Comment:
```suggestion
                                other => bail!("Unexpected contract error: {:?other}"),
```

I would also log the unexpected type of decoded error, just in case it ever pops up. it could be useful for debugging

(same suggestion for other commands that do this below)